pandas/core/indexes/period.py

```diff
@@ -704,36 +704,36 @@ def get_loc(self, key, method=None, tolerance=None):
         -------
         loc : int
         """
+        if is_integer(key):
+            ordinal = key
+        else:
```

Comment:
why don't you just move what you have in _cast_to_period_object here. This is confusing as you are doing the same type of operation in multiple functions.

Reply:
yah, i think we should require that the argument to get_loc be either a Period or a string/datetime that can be cast to Period, not allow ordinal at this point

Reply:
@jbrockmendel 
Yes, I really want to. But some code use ".get_loc" with numeric type and if exception should raise, raise KeyError not TypeError.